compiler/ast/ast_types.nim

```diff
@@ -1589,30 +1589,18 @@ type
     counter*: int
     data*: seq[PSym]
 
-  # -------------- backend information -------------------------------
-  TLocFlag* = enum
-    # XXX: `TLocFlag` conflates two things:
-    #      - flags regarding the external interface (e.g., `lfHeader`; set by
-    #        sem, used by sem and the code generators)
-    #      - location-related flags (e.g., ``lfIndirect``, ``lfSingleUse``,
-    #        etc.; only relevant to the C code generator)
-    #      Split the enum up.
-    lfIndirect,               ## backend introduced a pointer
-    lfFullExternalName, ## only used when 'conf.cmd == cmdNimfix': Indicates
+  ExternalFlag* = enum
```

Comment:
Could also call it `ExternFlag`, leaning a bit on `extern` being evocative of effecting the ABI or external binary interface.

---

Thought of this after, I think this is to do with how the symbol manifests in the intended compiled artifact(s)'. Sticking with that some alternatives are:

- `ArtifactFlag`
- `BinaryInterfaceFlag`

Personally, I like the existing `ExternalFlag` more than the above suggestions, merely noting them in case they inspire a better name and to track what we considered.

Finally, perhaps this flag is a misplaced symbol flag?

Reply:
Hmm, about it being a misplaced symbol flag, I'm not sure.

While a normal symbol flag is relevant to both semantic analysis and the backend (the flags modify what the symbol represents), `ExternalFlag`s are, for the largest part, only relevant to the code generators (they only affect code generation [^1]), so I *think* that having them separate makes sense, at least for now.

[^1]: not entirely; `exfNoDecl` and `exfDynamicLib` do have an effect on observable behaviour